api/v1beta1/hcpvaultsecretsapp_types.go

```diff
@@ -18,6 +18,8 @@ type HCPVaultSecretsAppSpec struct {
 	// Kubernetes namespace. HCPAuthRef string `json:"hcpAuthRef,omitempty"`
 	HCPAuthRef string `json:"hcpAuthRef,omitempty"`
 	// RefreshAfter a period of time, in duration notation e.g. 30s, 1m, 24h
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Pattern="^(-?[0-9]+(\\.[0-9]+)?(s|m|h))+$"
```

Comment:
I could be misreading, but a few minor suggestions on the regex:

* It seems like it would have accepted negative values before?
* The + on the end seems like it would allow the whole pattern to repeat, like 600s600s?
* `[.]` saves any headache around interpreting the level of escaping needed for backslashes.

```suggestion
	// +kubebuilder:validation:Pattern="^([0-9]+([.][0-9]+)?(s|m|h))$"
```

Reply:
+1 for the catch on the duration string duplication. I think we can probably omit the negative duration for now. I believe that was to support the case where no refresh could be configured e.g: `skipRefresh := d < 0`, but I think we can probably omit support for that. It is possible to set the min-refresh to 0s and then `spec.refreshAfter` could be set to 0s.

Reply:
Think we still need to escape '.' in the character class. I am inclined to leave it as is since it's the only character allowed.

E.g:
 Invalid value: "^(?[0-9]+([.][0-9]+)?(s|m|h))$": must be a valid regular expression

Reply:
Updated in c6929b4eb6bd75e59a86f5e4e68b5a3962f7275f